code/game/objects/items.dm

```diff
@@ -102,6 +103,18 @@
 	if(!gc_destroyed)
 		contents_explosion(severity, target)
 
+/obj/item/proc/burn()
+	src.visible_message("<span class='warning'>[src] disintegrates, leaving behind a pile of ashes.</span>")
+	new /obj/effect/decal/cleanable/ash(src.loc)
+	qdel(src)
+
+/obj/item/proc/extinguish()
+	if(burning)
+		burning = 0
+		for(var/image/img in overlays)

```

Comment:
PLEASE NO.

store a reference to the fire image (I assume there is only one) on the item, and simply remove it then null it.
bonus points if the fire image is generic enough and you define it as a global/static so only one image exists ever (if there is only one generic image you don't need to reference it on the item, since the global/static var itself is the reference)